salt/utils/process.py

```diff
@@ -774,31 +745,32 @@ def _run(self):
             # sys.stderr and set the proper exitcode and we have already logged
             # it above.
             raise
+        finally:
+            for method, args, kwargs in self._finalize_methods:
+                method(*args, **kwargs)
 
 
 class SignalHandlingMultiprocessingProcess(MultiprocessingProcess):
     def __init__(self, *args, **kwargs):
         super(SignalHandlingMultiprocessingProcess, self).__init__(*args, **kwargs)
-        if salt.utils.platform.is_windows():
-            if hasattr(self, '_is_child'):
-                # On Windows, no need to call register_after_fork().
-                # register_after_fork() would only work on Windows if called
-                # from the child process anyway. Since we know this is the
-                # child process, call __setup_signals() directly.
-                self.__setup_signals()
-        else:
-            multiprocessing.util.register_after_fork(
-                self,
-                SignalHandlingMultiprocessingProcess.__setup_signals
-            )
+        self._signal_handled = multiprocessing.Value('i', 0)
```

Comment:
Just curious, why a `Value` instead of for example `Event`?

Reply:
No particular reason. I can change this to event if you prefer.

Reply:
Nope, no need to change. Just though there could be a reason behind it, like performance or similar and I just wanted to know so I could also do better next time I was faced with a similar problem.